moquette/broker/src/main/java/io/moquette/broker/Session.java

```diff
@@ -365,13 +363,12 @@ private void sendPublishQos2(Topic topic, MqttQoS qos, ByteBuf payload) {
 
             // Retain before adding to map
             payload.retain();
-            EnqueuedMessage old = inflightWindow.put(packetId, new SessionRegistry.PublishedMessage(topic, qos, payload));
+            EnqueuedMessage old = addToInFlightWindow(packetId, new PublishedMessage(topic, qos, payload));
             // If there already was something, release it.
             if (old != null) {
                 old.release();
                 inflightSlots.incrementAndGet();
             }
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 360:383, 333:356.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.